### PR TITLE
cloud: buffer and put uploader for azure

### PR DESCRIPTION
**Pull request by stevendanna**
This is for a debugging experiment. Ideally we would not commit this. If we do, we must add some tests.

Epic: None

Release note: None

### CODE REVIEW

File: pkg/cloud/azure/azure_storage.go [hunk 1]
```diff
@@ -157,6 +171,36 @@ func (s *azureStorage) Writer(ctx context.Context, basename string) (io.WriteClo
...
+	b    *bytes.Buffer
+	sp   *tracing.Span
+}
+
+func (u *putBlobUploader) Write(p []byte) (int, error) {
+	return u.b.Write(p)
+}
+
+func (u *putBlobUploader) Close() error {
+	defer u.sp.Finish()
```
Comment (msbutler) [inline pkg/cloud/azure/azure_storage.go:186]: ^somebody needs to call this, no? Maybe in `(s *azureStorage) Close()`

Comment (stevendanna) [reply pkg/cloud/azure/azure_storage.go:186]: I believe the caller of Writer is responsible for closing the returned writer.

**Comment by stevendanna** [comment]
Ran this through the cloud test pipeline here: https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_CloudUnitTests/8216387?buildTab=tests

**Review by benbardin — APPROVED** [review]

**Comment by stevendanna** [comment]
Looking into a use of span after close. If I can't sort it out, we can always just remove the tracing span here.

**Comment by stevendanna** [comment]
bors r=benbardin

**Comment by stevendanna** [comment]
bors retry

**Comment by stevendanna** [comment]
bors retry

**Comment by shermanCRL** [comment]
Sounds like the backports went cold? I think a customer is waiting on them?

**Pull request merged**